iceoryx_posh/test/moduletests/test_roudi_process.cpp

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2020 by Robert Bosch GmbH. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "iceoryx_posh/iceoryx_posh_types.hpp"
+#include "iceoryx_posh/internal/roudi/roudi_process.hpp"
+#include "iceoryx_posh/mepoo/mepoo_config.hpp"
+#include "iceoryx_posh/roudi/memory/roudi_memory_interface.hpp"
+#include "iceoryx_posh/version/compatibility_check_level.hpp"
+#include "iceoryx_utils/cxx/string.hpp"
+#include "iceoryx_utils/platform/types.hpp"
+#include "test.hpp"
+
+using namespace ::testing;
+using namespace iox::roudi;
+using namespace iox::popo;
+using namespace iox::runtime;
+using ::testing::Return;
+
+class MqInterfaceUser_Mock : public iox::roudi::RouDiProcess
+{
+  public:
+    MqInterfaceUser_Mock()
+        : iox::roudi::RouDiProcess("TestRoudiProcess", 200, nullptr, true, 0x654321, 255)
+    {
+    }
+    MOCK_METHOD1(sendToMQ, void(MqMessage));
+};
+
+class RouDiProcess_test : public Test
+{
+  public:
+    const iox::ProcessName_t processname = "TestRoudiProcess";
+    pid_t pid{200U};
+    iox::mepoo::MemoryManager* payloadMemoryManager{nullptr};
+    bool isMonitored = true;
+    const uint64_t payloadSegmentId{0x654321U};
+    const uint64_t sessionId{255U};
+    MqInterfaceUser_Mock mqIntrfceusermock;
+};
+
+TEST_F(RouDiProcess_test, getPid)
+{
+    RouDiProcess roudiproc(processname, pid, payloadMemoryManager, isMonitored, payloadSegmentId, sessionId);
+    EXPECT_THAT(roudiproc.getPid(), Eq(200));
```

Comment:
Please use the `pid` member instead of the magic number. Please also do it in the other tests.

Reply:
done